src/flp/types/fixedpoint_l2.rs

```diff
@@ -153,16 +153,30 @@
 //! For example, it is `0.999969482421875` for `n = 16`.
 
 pub mod compatible_float;
+pub mod noise;
 
 use crate::field::{FftFriendlyFieldElement, FieldElementExt, FieldElementWithInteger};
 use crate::flp::gadgets::{BlindPolyEval, ParallelSumGadget, PolyEval};
 use crate::flp::types::fixedpoint_l2::compatible_float::CompatibleFloat;
+use crate::flp::types::fixedpoint_l2::noise::sample_discrete_gaussian;
 use crate::flp::{FlpError, Gadget, Type};
 use crate::polynomial::poly_range_check;
 use fixed::traits::Fixed;
+use num_bigint::{BigInt, BigUint, TryFromBigIntError};
 
 use std::{convert::TryFrom, convert::TryInto, fmt::Debug, marker::PhantomData};
 
+use self::noise::compute_std_deviation;
+
+/// The privacy parameter which is passed to `FixedPointBoundedL2VecSum` has this type.
+pub type PrivacyParameterType = (u128, u128);
```

Comment:
it is a bit awkward: those two integers simply encode a rational number. as we never need any complicated rational number operations, we didn't want to introduce a dependency, but i suppose we could just call this type `Fraction` of something? does it make sense to use a struct for this? the reason this is a type alias at all is so that we can include it in our packages calling prio. it should definitely be documented properly, sorry. working on that.

all functions used during sampling need to accept bigints as arguments because `u128` is insufficient for the scale we use. the gaussian sampler could accept `u128` and convert internally, but it can deal with bigint inputs so i thought it makes sense to allow them even if we don't actually use it here.